drivers/sensor/bosch/bmm350/bmm350.c

```diff
@@ -0,0 +1,984 @@
+			ret = bmm350_reg_write(dev, BMM350_REG_PMU_CMD, 0x00);
+		}
+	}
+	return ret;
+}
+/*!
+ * @brief This API is used to read uncompensated mag and temperature data.
+ */
+int8_t bmm350_read_uncomp_mag_temp_data(struct bmm350_raw_mag_data *raw_data,
+					const struct device *dev)
+{
+	struct bmm350_data *data = dev->data;
+	int8_t rslt = BMM350_OK;
+	uint8_t mag_data[14] = {0};
+	uint32_t raw_mag_x, raw_mag_y, raw_mag_z, raw_temp;
+
+	if (raw_data != NULL) {
+		/* Get uncompensated mag data */
+		rslt = bmm350_reg_read(dev, BMM350_REG_MAG_X_XLSB, mag_data, 14);
+
+		if (rslt == BMM350_OK) {
+			raw_mag_x = (uint32_t)mag_data[2] + ((uint32_t)mag_data[3] << 8) +
+				    ((uint32_t)mag_data[4] << 16);
+			raw_mag_y = (uint32_t)mag_data[5] + ((uint32_t)mag_data[6] << 8) +
+				    ((uint32_t)mag_data[7] << 16);
+			raw_mag_z = (uint32_t)mag_data[8] + ((uint32_t)mag_data[9] << 8) +
+				    ((uint32_t)mag_data[10] << 16);
+			raw_temp = (uint32_t)mag_data[11] + ((uint32_t)mag_data[12] << 8) +
+				   ((uint32_t)mag_data[13] << 16);
+
+			if ((data->axis_en & BMM350_EN_X_MSK) == BMM350_DISABLE) {
+				raw_data->raw_xdata = BMM350_DISABLE;
+			} else {
+				raw_data->raw_xdata = fix_sign(raw_mag_x, BMM350_SIGNED_24_BIT);
+			}
+
+			if ((data->axis_en & BMM350_EN_Y_MSK) == BMM350_DISABLE) {
+				raw_data->raw_ydata = BMM350_DISABLE;
+			} else {
+				raw_data->raw_ydata = fix_sign(raw_mag_y, BMM350_SIGNED_24_BIT);
+			}
+
+			if ((data->axis_en & BMM350_EN_Z_MSK) == BMM350_DISABLE) {
+				raw_data->raw_zdata = BMM350_DISABLE;
+			} else {
+				raw_data->raw_zdata = fix_sign(raw_mag_z, BMM350_SIGNED_24_BIT);
+			}
+
+			raw_data->raw_data_t = fix_sign(raw_temp, BMM350_SIGNED_24_BIT);
+		}
+	} else {
+		rslt = BMM350_E_NULL_PTR;
+	}
+
+	return rslt;
+}
+static int8_t read_out_raw_data(int32_t *out_data, const struct device *dev)
+{
+	int8_t rslt = BMM350_OK;
+	int32_t temp = 0;
+	struct bmm350_raw_mag_data raw_data = {0};
+
+	if (out_data != NULL) {
+		rslt = bmm350_read_uncomp_mag_temp_data(&raw_data, dev);
+
+		if (rslt == BMM350_OK) {
+			/* Convert mag lsb to uT and temp lsb to degC */
+			out_data[0] = (
+				(raw_data.raw_xdata * BMM350_LSB_TO_UT_XY_COEFF) /
+				BMM350_LSB_TO_UT_COEFF_DIV);
+			out_data[1] = (
+				(raw_data.raw_ydata * BMM350_LSB_TO_UT_XY_COEFF) /
+				BMM350_LSB_TO_UT_COEFF_DIV);
+			out_data[2] = (
+				(raw_data.raw_zdata * BMM350_LSB_TO_UT_Z_COEFF) /
+				BMM350_LSB_TO_UT_COEFF_DIV);
+			out_data[3] = (
+				(raw_data.raw_data_t *BMM350_LSB_TO_UT_TEMP_COEFF) /
+				BMM350_LSB_TO_UT_COEFF_DIV);
+
+			if (out_data[3] > 0) {
+				temp = (out_data[3] - (2549 / 100));
+			} else if (out_data[3] < 0) {
+				temp = (out_data[3] + (2549 / 100));
+			} else {
+				temp = out_data[3];
+			}
+
+			out_data[3] = temp;
+		}
+	} else {
+		rslt = BMM350_E_NULL_PTR;
+	}
+
+	return rslt;
+}
+
+int8_t bmm350_get_compensated_mag_xyz_temp_data_fixed(struct bmm350_mag_temp_data *mag_temp_data,
+						      const struct device *dev)
```

Comment:
nit: usually, I always see the `const struct device *` as the first arg for every function. Just something that seems to not follow a pattern